test/BreakInvariantBounty.test.js

```diff
@@ -17,82 +17,95 @@ const sendReward = async (from, to, value) => ethSendTransaction({
-      await sendReward(owner, this.bounty.address, reward);
-
-      const balance = await ethGetBalance(this.bounty.address);
-      balance.should.be.bignumber.equal(reward);
-    });
-
-    context('with reward', function () {
-      beforeEach(async function () {
-        const result = await this.bounty.createTarget({ from: researcher });
-        const event = expectEvent.inLogs(result.logs, 'TargetCreated');
-
-        this.targetAddress = event.args.createdAddress;
-
-        await sendReward(owner, this.bounty.address, reward);
-
-        const balance = await ethGetBalance(this.bounty.address);
-        balance.should.be.bignumber.equal(reward);
-      });
+contract('BreakInvariantBounty', function ([_, owner, researcher, anyone, nonTarget]) {
+  beforeEach(async function () {
+    this.bounty = await BreakInvariantBountyMock.new({ from: owner });
+  });
 
-      it('cannot claim reward', async function () {
-        await assertRevert(
-          this.bounty.claim(this.targetAddress, { from: researcher }),
-        );
-      });
-    });
+  it('can set reward', async function () {
+    await sendReward(owner, this.bounty.address, reward);
+    (await ethGetBalance(this.bounty.address)).should.be.bignumber.equal(reward);
   });
 
-  context('against broken contract', function () {
+  context('with reward', function () {
     beforeEach(async function () {
-      this.bounty = await InsecureInvariantTargetBounty.new();
-
-      const result = await this.bounty.createTarget({ from: researcher });
-      const event = expectEvent.inLogs(result.logs, 'TargetCreated');
-
-      this.targetAddress = event.args.createdAddress;
       await sendReward(owner, this.bounty.address, reward);
     });
 
-    it('can claim reward', async function () {
-      await this.bounty.claim(this.targetAddress, { from: researcher });
-      const claim = await this.bounty.claimed();
-
-      claim.should.equal(true);
-
-      const researcherPrevBalance = await ethGetBalance(researcher);
+    describe('destroy', function () {
+      it('returns all balance to the owner', async function () {
+        const ownerPreBalance = await ethGetBalance(owner);
+        await this.bounty.destroy({ from: owner, gasPrice: 0 });
+        const ownerPostBalance = await ethGetBalance(owner);
 
-      await this.bounty.withdrawPayments(researcher, { gasPrice: 0 });
-      const updatedBalance = await ethGetBalance(this.bounty.address);
-      updatedBalance.should.be.bignumber.equal(0);
+        (await ethGetBalance(this.bounty.address)).should.be.bignumber.equal(0);
+        ownerPostBalance.sub(ownerPreBalance).should.be.bignumber.equal(reward);
+      });
 
-      const researcherCurrBalance = await ethGetBalance(researcher);
-      researcherCurrBalance.sub(researcherPrevBalance).should.be.bignumber.equal(reward);
+      it('reverts when called by anyone', async function () {
+        await assertRevert(this.bounty.destroy({ from: anyone }));
```

Comment:
What about s/anyone/notOwner ?

Reply:
Maybe, I'm not really a fan of `anyone`, but it's _very_ widely used throughout the codebase, we'd have to change it everywhere.

Reply:
let's make it a good first issue for hacktoberfest.

Reply:
Created https://github.com/OpenZeppelin/openzeppelin-solidity/issues/1357